cdap-app-templates/cdap-etl/hydrator-spark-core-base/src/main/java/io/cdap/cdap/etl/spark/batch/SparkBatchSinkFactory.java

```diff
@@ -173,9 +165,34 @@ private void addOutput(String stageName, String datasetName, String alias, Map<S
         sec.saveAsDataset(rdd, datasetInfo.getDatasetName(), datasetInfo.getDatasetArgs());
       }
     }
+
+    if (outputFormats.size() == 1) {
+      saveUsingOutputFormat(outputFormats.values().iterator().next(), rdd);
+    }
```

Comment:
Shouldn't we return after calling `saveUsingOutputFormat`?

Reply:
agh yes